### PR TITLE
LUN-130: Zoom in photo with react-native-image-zoom-viewer

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/components/ImageCarousel.tsx [hunk 1]
```diff
@@ -1,56 +1,75 @@
...
 }
 
 const ImageCarousel = ({ images }: ImageCarouselPropsType): ReactElement => {
-  const [activeImage, setActiveImage] = useState(0)
-  const { width: viewportWidth } = useWindowDimensions()
+  const imagesUrls: ImageUrlType[] = images.map(image => ({
+    url: image.image
+  }))
+
+  const { height: viewportHeight } = useWindowDimensions()
```
Comment (Taggotty) [inline src/components/ImageCarousel.tsx:43]: can be removed?

File: src/components/ImageCarousel.tsx [hunk 2]
```diff
@@ -1,56 +1,75 @@
...
+
+  const renderIndicator = (currentIndex?: number, allSize?: number): JSX.Element => {
```
Comment (Taggotty) [inline src/components/ImageCarousel.tsx:45]: Just personal preference.

```suggestion
  const renderIndicator = (currentIndex?: number, allSize?: number): ReactElement => {
```

File: src/routes/vocabulary-list/components/__tests__/VocabularyListModal.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import React from 'react'
+import VocabularyListModal from '../VocabularyListModal'
+import { DocumentsType } from '../../../../constants/endpoints'
+import { render, fireEvent } from '@testing-library/react-native'
+import labels from '../../../../constants/labels.json'
+
+jest.mock('../../../../components/AudioPlayer', () => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (Taggotty) [inline src/routes/vocabulary-list/components/__tests__/VocabularyListModal.spec.tsx:8]: Instead you can add the following to the eslintrc.js overrides
```javascript
    {
      files: ['*.spec.{ts,tsx}'],
      rules: {
        '@typescript-eslint/no-empty-function': 'off',
        '@typescript-eslint/no-var-requires': 'off'
      }
    }
```

**Review by Taggotty — APPROVED** [review]
I personally think the caroussel is sometimes barely visible. Maybe it is better to display the dots below the images. But that can also be an issue for later.

- Tested on Android Emulator

File: src/routes/vocabulary-list/components/VocabularyListItem.tsx [hunk 4]
```diff
@@ -69,13 +69,14 @@ export const styles = StyleSheet.create({
...
+  setIsModalVisible?: () => void
 }
 
-const VocabularyListItem = ({ document }: VocabularyListItemPropType): ReactElement => {
+const VocabularyListItem = ({ document, setIsModalVisible }: VocabularyListItemPropType): ReactElement => {
   const { article, word } = document
 
   return (
-    <View style={styles.wrapper}>
+    <Pressable style={styles.wrapper} onPress={setIsModalVisible ? () => setIsModalVisible() : () => {}}>
```
Comment (klinzo) [inline src/routes/vocabulary-list/components/VocabularyListItem.tsx:79]:
```suggestion
    <Pressable style={styles.wrapper} onPress={setIsModalVisible ?? () => {}}>
```

**Review by klinzo — APPROVED** [review]
Tested on Android

**Pull request merged**